oeps/architectural-decisions/oep-0058-arch-translations-management.rst

```diff
@@ -0,0 +1,263 @@
+openedx/translations repository. This link will allow for the `Transifex GitHub App`_ to
+automatically manage the push/pull of the translation files without the need for human
+intervention.
+
+.. _edx-platform: https://github.com/openedx/edx-platform
+.. _openedx-translations: https://github.com/openedx/openedx-translations
+
+Add `Transifex GitHub App`_ to openedx Organization
+===================================================
+
+The `Transifex GitHub App`_ will need to be added to the openedx GitHub organization in
+order to grant the app permissions to push/pull the translation files. Currently, we
+manage the push/pull permissions for the edx-transifex-bot through a number of GitHub
+user groups. The `Transifex GitHub App`_ once installed in an organization, is granted
+permissions to push/pull on a repository basis, and by moving all the translation files
+to a single repository we eliminate separate translations user groups.
+
+Connect the New Translation Repository to Transifex
+===================================================
+
+The Transifex web-app accepts a `Transifex GitHub Integration configuration file`_ for
+each Transifex project. By connecting the single repository containing all translation
+files, we only need to make a single `Transifex GitHub Integration configuration file`_
+that allows the `Transifex GitHub App`_ to manage the translation files. Based on the
+Translation Working Group's instruction on acceptable translation/review percentages, we
+can set parameters that automatically push and pull translation files.
+
+Copy Transifex's Translation Memory and Combine Translators
+===========================================================
+
+As a last step we will reorganize the openedx Transifex organization by combining
+translators and reviewers across Transifex projects into the new project associated with
+the new repository. In addition, we can save all the progress the Open edX translators
+have accomplished by copying the Transifex's Translation Memory, the auto-translation
+feature that allows for Projects with similar strings to be automatically translated,
+from the old projects to this new one. Once older projects are made redundant by the new
+project, they will be deprecated. By moving all the translation files to the same
+repository we can increase the reach of the Transifex's Translation Memory feature to
+help translate similar strings across the entire code/documentation base.
+
+Get Translations Back for Deployment/Development
+================================================
+
+A new python library, called openedx-atlas, will be created. This will enable the
+placement of the translation files kept in openedx-translations into locally cloned
+repositories for development and containers containing the code translation files are
+formed from. This tool will manage the placement of translation files through an editable
+atlas configuration file (atlas.yml) kept in the repositories that have
```

Comment:
openedx-atlas already provides the ability to override default parameter values and an atlas.yml configuration file by using the CLI's flags. 

`atlas pull -r "openedx/openedx-translations" -b "main" -d "credentials"`

This would pull translation files from the credentials directory in the main branch 
 from the repository openedx/openedx-translations.

There are many other features that could be beneficial, such as the ones you described in your comment. We can hash out these implementation details further in [Follow-up Work](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3578494977/OEP-58+Translations+Management).

Reply:
semi-related thread: https://github.com/openedx/open-edx-proposals/pull/367#discussion_r1023212567